### PR TITLE
Sort the inventory

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #1799 

Depends: #1798 
Depends: #1797

### CODE REVIEW

**Comment by fmatthew5876** [comment]
Rebased to maste

File: src/game_party.cpp [hunk 1]
```diff
@@ -707,3 +700,9 @@ Game_Actor* Game_Party::GetHighestLeveledActorWhoCanUse(const RPG::Item* item) c
 	}
 	return best;
 }
+
+std::pair<int,bool> Game_Party::GetItemIndex(int item_id) const {
+	auto& ids = data().item_ids;
+	auto iter = std::lower_bound(ids.begin(), ids.end(), item_id);
+	return std::make_pair(iter - ids.begin(), (iter != ids.end() && *iter == item_id));
```
Comment (Ghabry) [inline src/game_party.cpp]: Why do you use std::lower_bound and not std::find here when the goal is to find the item with the specific id.

Comment (fmatthew5876) [reply src/game_party.cpp]: lower_bound does a binary search, which find the item in lg(N) time instead of linear time for find.

Comment (Ghabry) [reply src/game_party.cpp]: and std::binary_search returns bool instead of an iterator, hmpf.

Comment (fmatthew5876) [reply src/game_party.cpp]: Oh yeah, that one is kind of useless.. Unless all you want to do is ask "Does this thing exist in the set"? Which is rare.

**Pull request merged**